endpoints/cookie_sync_test.go

```diff
@@ -2207,3 +2274,145 @@ func getDefaultActivityConfig(componentName string, allow bool) *config.AccountP
 		},
 	}
 }
+
+func TestSetCookieDeprecationHeader(t *testing.T) {
+	getTestRequest := func(addCookie bool) *http.Request {
+		r := httptest.NewRequest("POST", "/cookie_sync", nil)
+		if addCookie {
+			r.AddCookie(&http.Cookie{Name: receiveCookieDeprecation, Value: "1"})
+		}
+		return r
+	}
+
+	type args struct {
+		w       http.ResponseWriter
+		r       *http.Request
+		account *config.Account
+	}
+	tests := []struct {
+		name                            string
+		args                            args
+		expectedCookieDeprecationHeader bool
+	}{
+		{
+			name: "receive-cookie-deprecation not present in request but account is nil",
```

Comment:
When using t.Run, test names should be shorter and not include spaces. Consider a more concise name to identify the test and use a comment on the line to provide a more detailed description as needed. For example, this test case might be called something like "not-present-account-nil".

Reply:
done